.github/workflows/e2e-npm-dep-check.yaml

```diff
@@ -0,0 +1,86 @@
+name: "E2E NPM Dependency Updates"
+
+on:
+  schedule:
+    - cron: "0 16 * * 2" # Every Tuesday at 8:00-9:00am Pacific Time.
+  workflow_dispatch:
+
+# START HERE! v
+
+# THE ONLY EDITING NEEDED IN THIS FILE IS THE ENV VARS BELOW.
+# By default, PATH_TO_PACKAGE_JSON is root level.
+
+env:
+  PATH_TO_PACKAGE_JSON: "src/e2e" # Directory of package.json relative to root.
+  ISSUE_TITLE: "e2e NPM Dependency Updates" # Title of Issue displayed in Issues tab in GitHub.
+
+jobs:
+  check-versions:
+    runs-on: ubuntu-22.04
+    container:
+      # Lightweight NodeJS Image - v20.2.x
+      image: node:20.5-bullseye-slim
+
+    steps:
+      # Checkout branch.
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Run NodeJS script to check for latest npm dependency versions and capture output.
+      - name: Run NPM DEP Check Node.js script
+        id: check_versions
+        env:
+          PACKAGE_JSON: ${{ env.PATH_TO_PACKAGE_JSON }}
+        run: |
+          node .github/helpers/check-npm-dependencies.js > output.txt
+
+      # Upload the output as an artifact.
+      - name: Upload output
+        uses: actions/upload-artifact@v3
+        with:
+          name: output
+          path: output.txt
+
+  create-issue:
+    needs: check-versions
+    runs-on: ubuntu-22.04
+    container:
+      # Lightweight NodeJS Image - v20.2.x
+      image: node:20.2-bullseye-slim
+
+    steps:
+      # Download the output artifact.
+      - name: Download output
+        uses: actions/download-artifact@v3
+        with:
+          name: output
+          path: .
+
+      # Encode and set ISSUE_BODY env.
+      - name: Set ISSUE_BODY env
+        shell: bash
+        run: |
+          OUTPUT=$(cat output.txt)
+          OUTPUT="${OUTPUT//'%'/'%25'}"
+          OUTPUT="${OUTPUT//$'\n'/'%0A'}"
+          OUTPUT="${OUTPUT//$'\r'/'%0D'}"
+          echo "ISSUE_BODY=${OUTPUT}" >> $GITHUB_ENV
+
+      # Checkout branch.
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

Comment:
Note to self: Update to v4

Reply:
all the other workflows need to be updated too...

Reply:
It's nothing urgent but whenever we update a workflow we should also be updating the action versions